File: OOP/1_intro/Reckless+Driver+-+C++/RecklessDriver/RecklessDriver/FireHydrant.cpp
#include "FireHydrant.h"
#include "Player.h"
#include <iostream>
namespace RecklessDriver {
	FireHydrant::FireHydrant(int _damage, int _cash):SideObject(_damage, _cash)
	{
		SetName("FireHydrant");
	}


	FireHydrant::~FireHydrant()
	{
	}

	void FireHydrant::OnCollision(GameObject *other)
	{
		if (other->GetTag() == "Player") {
			Player *p = (Player *)other;
			if (count == 0) {
				std::cout << "#### COLLISION ->[FireHydrant] Fountain" << std::endl; 
				p->ApplyDamage(GetDamage(), GetCash());
			}
			else {
				std::cout << "#### COLLISION ->[FireHydrant] Collided again" << std::endl;
				p->ApplyDamage(GetDamage(), GetCash() * count);
			}
		}
	}
}

File: OOP/1_intro/Reckless+Driver+-+Java/Reckless Driver/src/com/poash/rd/FireHydrant.java
package com.poash.rd;

public class FireHydrant extends SideObject {
	public FireHydrant(int damage, int cash) {
		super(damage, cash);
		setName("FireHydrant");
	}

	@Override
	public void OnCollision(GameObject other) {
		if(other.getTag().equals("Player")){
			Player player = (Player)other; 
			if(count == 0){
				System.out.println("#### COLLISION ->[FireHydrant] Fountain");
				player.ApplyDamage(getDamage(), getCash());
			}else{
				System.out.println("#### COLLISION ->[FireHydrant] Collided again");
				player.ApplyDamage(getDamage(), getCash() * count);
			}
		}
	}
}
